core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -597,10 +595,6 @@ protected StructReader(List<ValueReader<?>> readers, Types.StructType struct, Ma
 
     protected abstract void set(S struct, int pos, Object value);
 
-    protected Object prepareConstant(Type type, Object value) {
```

Comment:
I'm moving this out of Avro and adding a callback to convert the constants to `PartitionUtil.constantsMap`. That way, Spark can supply a conversion function and use it in both places, instead of duplicating the conversion in Avro and Parquet readers.